### PR TITLE
Sc 7490 replace chief

### PR DESCRIPTION
**Pull request by gbalabasquer**
# Description

Spell Description:

COB Team Name or Author(s):

# Contribution Checklist

- [x] First commit title starts with COB Team Name and Collateral type (e.g. `___-TOKEN-X`)
- [x] Code approved
- [x] Tests approved
- [x] CI Tests pass

# Checklist

- [x] Every contract variable/method declared as public/external private/internal
- [x] Verify expiration (`4 days + 2 hours` monthly and `30 days` for the rest)
- [x] Verify hash in the description matches [here](https://emn178.github.io/online-tools/keccak_256.html)
- [x] Validate all addresses used are in Kovan changelog
- [x] Deploy spell to kovan `SOLC_FLAGS="--optimize --optimize-runs=1" dapp --use solc:0.5.12 build --extract && dapp create DssSpell --gas=5000000 --gas-price="$(seth --to-wei 3 "gwei")"`
- [x] Ensure contract is verified on `kovan` etherscan
- [x] Change test to use kovan spell address and deploy timestamp
- [x] Keep `Kovan-DssSpell.sol` and `Kovan-DssSpell.t.sol` the same, but make a copy in `archive`
- [ ] `squash and merge` this PR

### CODE REVIEW

**Comment by gbalabasquer** [comment]
This PR is still a draft.

**Comment by gbalabasquer** [comment]
PR ready to be checked.

File: src/Kovan-DssSpell.t.sol
Comment (lucas-manuel) [inline src/Kovan-DssSpell.t.sol:703]: Should add `assertTrue(!newChief.isUserRoot(address(spellTestMoms)));`

File: src/Kovan-DssSpell.t.sol
Comment (lucas-manuel) [inline src/Kovan-DssSpell.t.sol:714]: Should add `assertTrue(newChief.isUserRoot(address(spellTestMoms)));`

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
This LGTM, two small changes requested.

File: src/Kovan-DssSpell.t.sol
Comment (godsflaw) [inline src/Kovan-DssSpell.t.sol:17]: Once this PR is merged, we can merge:
https://github.com/makerdao/dss-interfaces/pull/38

**Review by godsflaw — APPROVED** [review]
LGTM so far, but we should have a discussion about making the new chief authority over SCD and adding some of those tests here, or if we feel comfortable abandoning SCD.  Did anyone audit the SCD Mom to see what actions one can still perform, and if that would impact SAI users redemption or minting of new SAI?

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:31]: Should we just remove that comment now that we are using the Chainlog?

Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol:31]: Honestly I like having that link there and I do click it when I'm reviewing.

**Comment by gbalabasquer** [comment]
I was leaving the SCD change just for mainnet, but makes sense we also change it in kovan.
I'd vote just to migrate the authority to the new chief and avoid any possible problem.

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:31]: Hmmm ok, question, should we link to 1.1.5 or to the new 1.1.6 one?

Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol:31]: Maybe just link to changelog.makerdao.com is fine.

File: src/Kovan-DssSpell.sol
Comment (brianmcmichael) [inline src/Kovan-DssSpell.sol:45]: the address is already a constant here `VOTE_PROXY_FACTORY`

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:31]: I'm linking to https://changelog.makerdao.com/releases/kovan/active/contracts.json

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol]: Changed!

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:31]: Done.

File: src/Kovan-DssSpell.t.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:703]: Done!

File: src/Kovan-DssSpell.t.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:715]: Done!

**Comment by gbalabasquer** [comment]
I've added the authority change of SCD to this spell.
Other option is setting them to address(0), which eventually we will need to do as `DssGov` doesn't work as an authority.

File: src/Kovan-DssSpell.sol
Comment (brianmcmichael) [inline src/Kovan-DssSpell.sol:38]: We should create a constructor for the spell action that sets these at deployment time, rather than execution time.

Comment (brianmcmichael) [reply src/Kovan-DssSpell.sol:38]: nvm, won't work in 0.5.12

File: src/Kovan-DssSpell.t.sol
Comment (lucas-manuel) [inline src/Kovan-DssSpell.t.sol:608]: I think its worth putting this repeated logic in a function.

Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:608]: I almost did that, let me see how it looks.

File: src/Kovan-DssSpell.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:30]: Do we even need this now that all addresses are being pulled from the on-chain changelog?

File: src/Kovan-DssSpell.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:40]:
```suggestion
        address MCD_PAUSE   = CHANGELOG.getAddress("MCD_PAUSE");
        address FLIPPER_MOM = CHANGELOG.getAddress("FLIPPER_MOM");
        address OSM_MOM     = CHANGELOG.getAddress("OSM_MOM");
```

File: src/Kovan-DssSpell.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:32]:
```suggestion
    ChainlogAbstract constant CHANGELOG = ChainlogAbstract(0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F);
```

File: src/Kovan-DssSpell.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.sol:35]:
```suggestion
    address constant MCD_ADM            = 0x27E0c9567729Ea6e3241DE74B3dE499b7ddd3fe6;
    address constant VOTE_PROXY_FACTORY = 0xEec40383cbeE179Be76D4D930c0ABD0D4beA672f;
```

File: src/Kovan-DssSpell.t.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.t.sol:744]: Rather than `SpellTest`, I'd call it `TestSpell` (the former is a little confusing since in our naming conventions it sounds like a contract that contains tests, rather than a mock contract created for testing).

File: src/Kovan-DssSpell.t.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.t.sol:767]: Similar comment here.

File: src/Kovan-DssSpell.t.sol
Comment (kmbarry1) [inline src/Kovan-DssSpell.t.sol:672]:
```suggestion
```

**Review by kmbarry1 — COMMENTED** [review]
LGTM modulo formatting nits. I think setting SAI authorities to zero is fine.

File: src/Kovan-DssSpell.t.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol:744]: Agreed, changing it.

File: src/Kovan-DssSpell.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.sol:30]: Brian preferred to keep it: https://github.com/makerdao/spells-kovan/pull/34#discussion_r528986005

File: src/Kovan-DssSpell.t.sol
Comment (gbalabasquer) [reply src/Kovan-DssSpell.t.sol]: I'm finally leaving it as it is, the grouping itself is a bit weird as the functions do different kind of stuff.
I've also added another function that does a similar thing but not exactly the same, so it would still leave that one out of a grouping function.

**Comment by gbalabasquer** [comment]
I've added Kurt's suggestions and an extra test for VoteProxy. I think is ready to be deployed.

**Review by kmbarry1 — APPROVED** [review]
LGTM

File: src/Kovan-DssSpell.sol
Comment (godsflaw) [inline src/Kovan-DssSpell.sol:40]: Since we removed the deployer EOA address from the changelog, I thought I would re-validate these changelog addresses for mainnet:

```
seth call 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F 'getAddress(bytes32)(address)' $(seth --to-bytes32 "$(seth --from-ascii "MCD_PAUSE")")
0xbE286431454714F511008713973d3B053A2d38f3
seth call 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F 'getAddress(bytes32)(address)' $(seth --to-bytes32 "$(seth --from-ascii "FLIPPER_MOM")")
0xc4bE7F74Ee3743bDEd8E0fA218ee5cf06397f472
seth call 0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F 'getAddress(bytes32)(address)' $(seth --to-bytes32 "$(seth --from-ascii "OSM_MOM")")
0x76416A4d5190d071bfed309861527431304aA14f
```

All good.  💪

**Review by godsflaw — APPROVED** [review]
LGTM

**Review by lucas-manuel — APPROVED** [review]

**Review by lucas-manuel — APPROVED** [review]
LGTM

**Pull request merged**
